src/engine.js

```diff
@@ -0,0 +1,62 @@
+import { gameStarter } from "./game";
+import "./style.scss";
+import Rectangle from "./rectangle";
+import { FRAME_DURATION } from "./constants";
+import { activate } from "./controls";
+
+let newRect = new Rectangle(0, 0, 100, 100);
+
+let canvas = document.querySelector("canvas");
+let ctx = canvas.getContext("2d");
+
+//for the game loop
+let prev = 0;
+let accumulatedProgress = 0;
+
+// setting the dimensions
+let [worldWidth, worldHeight] = [window.innerWidth, window.innerHeight];
+[canvas.width, canvas.height] = [worldWidth, worldHeight];
+
+gameStarter.then((value) => {
+  /**
+   * The **frame** function runs the game at a constant
+   * frame rate
+   *
+   * @param {number} timestamp Is a long integer value,
+   * the request id, that uniquely identifies the entry in the
+   * callback list.
+   */
+  let frame = (timestamp) => {
```

Comment:
Move variables related to the requestAnimationFrame loop away from the module scope
```suggestion
// setting the dimensions
let [worldWidth, worldHeight] = [window.innerWidth, window.innerHeight];
[canvas.width, canvas.height] = [worldWidth, worldHeight];

gameStarter.then((value) => {
  let prev = 0;
  let accumulatedProgress = 0;
  /**
   * The **frame** function runs the game at a constant
   * frame rate
   *
   * @param {number} timestamp Is a long integer value,
   * the request id, that uniquely identifies the entry in the
   * callback list.
   */
  let frame = (timestamp) => {
```